.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,159 @@
+      shell: 'bash'
+      if: |
+        !contains(fromJSON('["tar", "zip", "tgz", ""]'), inputs.compression_method)
+      env:
+        COMPRESSION_METHOD: '${{ inputs.compression_method }}'
+      run: |
+        echo "::error ::invalid compression_method '${COMPRESSION_METHOD}', allowed values: tar, zip, tgz"
+        exit 1
+
+    - name: 'Cache path'
+      id: 'cache-path'
+      if: |
+        contains(fromJSON('["true", "True", "TRUE", "1", "T", "t"]'), inputs.enable_cache)
+      shell: 'bash'
+      env:
+        NAME: '${{ inputs.name }}'
+        OVERRIDE_CACHE_KEY: '${{ inputs.override_cache_key }}'
+      run: |
+        CACHE_KEY="${RUNNER_OS}_${RUNNER_ARCH}_${NAME}"
+
+        if [ ! -z "${OVERRIDE_CACHE_KEY}" ]; then
+          CACHE_KEY="${OVERRIDE_CACHE_KEY}"
+        fi
+
+        echo "cache-key=${CACHE_KEY}" >> $GITHUB_OUTPUT
+
+    - name: 'Restore cache binary'
+      id: 'cache-binary'
+      if: |
+        contains(fromJSON('["true", "True", "TRUE", "1", "T", "t"]'), inputs.enable_cache)
+      uses: 'actions/cache/restore@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8' # ratchet:actions/cache/restore@v3
+      with:
+        path: '${{ inputs.install_dir }}'
+        key: '${{ steps.cache-path.outputs.cache-key }}'
+
+    - name: 'Create install dir'
+      if: |
+        steps.cache-binary.outputs.cache-hit != 'true'
+      shell: 'bash'
+      env:
+        INSTALL_DIR: '${{ inputs.install_dir }}'
+      run: |-
+        if [ ! -d "${INSTALL_DIR}" ]; then
+          mkdir "${INSTALL_DIR}"
+        fi
+
+    - name: 'Download binary'
+      if: |
+        steps.cache-binary.outputs.cache-hit != 'true'
+      shell: 'bash'
+      env:
+        INSTALL_DIR: '${{ inputs.install_dir }}'
+        DOWNLOAD_URL: '${{ inputs.download_url }}'
+      run: |-
+        cd "${INSTALL_DIR}"
+        curl -sSL -o "setup-binary-artifact" "${DOWNLOAD_URL}"
+
+    - name: 'Extract'
+      if: |
+        steps.cache-binary.outputs.cache-hit != 'true' &&
+        inputs.compression_method != ''
+      shell: 'bash'
+      env:
+        INSTALL_DIR: '${{ inputs.install_dir }}'
+        COMPRESSION_METHOD: '${{ inputs.compression_method }}'
+      run: |-
+        cd "${INSTALL_DIR}"
+
+        if [[ "${COMPRESSION_METHOD}" = "tar" ]]; then
+          tar -xvf setup-binary-artifact
+        elif [[ "${COMPRESSION_METHOD}" = "tgz" ]]; then
+          tar -xvzf setup-binary-artifact
+        elif [[ "${COMPRESSION_METHOD}" = "zip" ]]; then
+          unzip -o setup-binary-artifact
+        fi
+
+    - name: 'Save cache binary'
+      if: |
+        contains(fromJSON('["true", "True", "TRUE", "1", "T", "t"]'), inputs.enable_cache) &&
+        steps.cache-binary.outputs.cache-hit != 'true'
+      uses: 'actions/cache/save@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8' # ratchet:actions/cache/save@v3
+      with:
+        path: '${{ inputs.install_dir }}'
+        key: '${{ steps.cache-binary.outputs.cache-primary-key }}'
```

Comment:
Is this right? `cache-primary-key`?

Reply:
See example: https://github.com/actions/cache#using-a-combination-of-restore-and-save-actions

Reply:
replaced with value from inputs